hedera-cryptography/hedera-cryptography-tss/src/main/java/com/hedera/cryptography/tss/api/TssParticipantDirectory.java

```diff
@@ -0,0 +1,360 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.cryptography.tss.api;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.cryptography.pairings.signatures.api.PairingPrivateKey;
+import com.hedera.cryptography.pairings.signatures.api.PairingPublicKey;
+import com.hedera.cryptography.pairings.signatures.api.SignatureSchema;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.IntStream;
+
+/**
+ * Represents a directory of participants in a Threshold Signature Scheme (TSS).
+ *<p>Each participant has an associated id (called {@code participantId}), shares count and a tss encryption public key.
+ * It is responsibility of the user to assign each participant with a different deterministic integer representation.</p>
+ *
+ *<p>The current participant is represented by a {@code self} entry, and includes {@code participantId}'s id and the tss decryption private key.</p>
+ *<p>The expected {@code participantId} is the unique {@link Integer} identification for each participant executing the scheme.</p>
+ * <pre>{@code
+ * PairingPrivateKey tssDecryptionPrivateKey = ...;
+ * List<PairingPublicKey> tssEncryptionPublicKeys = ...;
+ * TssParticipantDirectory participantDirectory = TssParticipantDirectory.createBuilder()
+ *     //id, tss private decryption key
+ *     .self(0, persistentParticipantKey)
+ *     //id, number of shares, tss public encryption key
+ *     .withParticipant(0, 5, tssEncryptionPublicKeys.get(0))
+ *     .withParticipant(1, 2, tssEncryptionPublicKeys.get(1))
+ *     .withParticipant(2, 1, tssEncryptionPublicKeys.get(2))
+ *     .withParticipant(3, 1, tssEncryptionPublicKeys.get(3))
+ *     .withThreshold(5)
+ *     .build(signatureScheme);
+ * }</pre>
+ *
+ */
+public final class TssParticipantDirectory {
+    /**
+     * A sorted list of unique integer representations of each participant in the protocol.
+     */
+    private final List<Integer> sortedParticipantIds;
```

Comment:
there are several private members that aren't ever accessed. Should they be given getters?

Reply:
As this task aimed to expose the public interface needed for the lib, I intended to give getters used internally once the internal implementation uses them. But i can create them now if it makes sense.